aws/nat_gateway.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/retry"
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+)
+
+func getAllNatGateways(session *session.Session, excludeAfter time.Time, configObj config.Config) ([]*string, error) {
+	svc := ec2.New(session)
+
+	allNatGateways := []*string{}
+	input := &ec2.DescribeNatGatewaysInput{}
+	err := svc.DescribeNatGatewaysPages(
+		input,
+		func(page *ec2.DescribeNatGatewaysOutput, lastPage bool) bool {
```

Comment:
Oh, interesting, is this some new way to handle pagination in the new AWS Go SDK?

Reply:
I recall seeing this for a while, so I don't think it's new. However, it's not available in all API functions. It seems to be one of those things that are sometimes supported and othertimes not, depending on the team that manages the service.

Reply:
Roger